src/v/wasm/tests/wasm_allocator_test.cc

```diff
@@ -96,6 +99,34 @@ TEST(HeapAllocatorTest, CanReturnMemoryToThePool) {
     EXPECT_FALSE(mem.has_value());
 }
 
+MATCHER(HeapIsZeroed, "is zeroed") {
+    std::span<uint8_t> d = {arg.data.get(), arg.size};
+    for (auto e : d) {
+        if (e != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+TEST(HeapAllocatorTest, MemoryIsZeroFilled) {
+    size_t page_size = ::getpagesize();
+    heap_allocator allocator(heap_allocator::config{
+      .heap_memory_size = page_size,
+      .num_heaps = 1,
+    });
+    heap_allocator::request req{.minimum = page_size, .maximum = page_size};
+    auto allocated = allocator.allocate(req);
+    ASSERT_TRUE(allocated.has_value());
+    EXPECT_THAT(allocated, Optional(HeapIsZeroed()));
+    std::fill_n(allocated->data.get(), 4, 1);
+    allocator.deallocate(*std::move(allocated), 4);
+
+    allocated = allocator.allocate(req);
+    ASSERT_TRUE(allocated.has_value());
+    EXPECT_THAT(allocated, Optional(HeapIsZeroed()));
```

Comment:
question: what's the purpose of this second allocation?

Reply:
It's reused, so its ensured memory is zeroed when returned "dirty" to the allocator.

Reply:
ah i see